File: src/main/java/org/chousik/collection/Country.java
package org.chousik.collection;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.stream.Collectors;

/**
 * Класс перечисление стран
 */
public enum Country {
    RUSSIA,
    UNITED_KINGDOM,
    GERMANY,
    VATICAN;

    /**
     * @return возвращает все значения перечисления
     */
    public static String getValue() {
        return Arrays.stream(Country.values())
                .map(Country::name)
                .collect(Collectors.joining(", "));
    }
}

File: src/main/kotlin/collection/Country.kt
package org.chousik.collection

import java.util.*
import java.util.stream.Collectors

enum class Country {
    RUSSIA,
    UNITED_KINGDOM,
    GERMANY,
    VATICAN;

    companion object {
        val value: String
            get() = Arrays.stream(entries.toTypedArray())
                .map { obj: Country -> obj.name }
                .collect(Collectors.joining(", "))
    }
}